esp-hal/src/gpio/mod.rs

```diff
@@ -909,6 +953,14 @@ where
     fn clear_interrupt(&mut self, _: private::Internal) {
         <Self as GpioProperties>::Bank::write_interrupt_status_clear(1 << (GPIONUM % 32));
     }
+
+    fn set_interrupt_handler(&mut self, handler: Option<fn()>, _: private::Internal) {
+        if let Some(handler) = handler {
+            USER_PIN_HANDLERS[self.number(private::Internal) as usize].store(handler);
```

Comment:
No, the idea is that reading the pointer should be cheap, and I'm hoping AtomicPtr just gives us a single 32bit load.

Reply:
This has been removed, but a similar pattern existed in the "global" user handler. The critical section is unnecessary because the function requires exclusive access. The atomic pointer load should be cheaper than a critical section at least on multi-core chips, and I'm hoping we want cheap GPIO interrupt handlers, especially when the user handler has footgun-level limitations and probably has limited usefulness.